src/sqlfluff/rules/L036.py

```diff
@@ -74,8 +74,7 @@ def _eval(self, context: RuleContext):
             if len(select_targets_info.select_targets) == 1:
                 return self._eval_single_select_target_element(
                     select_targets_info,
-                    context.functional.segment,
-                    context.parent_stack,
+                    context,
```

Comment:
Rule L036 required heavy rework because it was previously leaving the parse tree in a pretty bad state. It was leaving trailing whitespace and comment child segments after deleting or moving other segments. To make the parser happy, these segments had to be relocated. The new `_fixes_for_move_after_select_clause()` function assists with this.